src/parseTools.mjs

```diff
@@ -262,14 +262,32 @@ function makeInlineCalculation(expression, value, tempVar) {
 
 // XXX Make all i64 parts signed
 
+function castToBigInt(x) {
+  // If given string is already a "BigInt(something)", then no-op.
+  if (typeof x === 'string' && x.startsWith('BigInt(')) return x;
+  // Or if it is already a BigInt literal string, e.g. "42n".
+  if (typeof x === 'string' && x.endsWith('n')) {
+    var maybeBigIntLiteral = Number(x.slice(0, -1));
+    if (Number.isInteger(maybeBigIntLiteral) && isFinite(maybeBigIntLiteral)) return x;
+  }
```

Comment:
Do these cases actually happen in the codebase?   should we just assert that they cannot happen?   I worry that if they are not happening in our codebase then these codepaths are untested/unused.

Reply:
In this PR I changed one test to do

```
{{{ makeSetValue('ptr', '0', 0x1122334455667788AAn, 'i64') }}};
```
and I figured that it is best to be consistent and not leave it to a half-way house (or if I wasn't, I'd get reviewed with a bunch of "what if user passes BigInt(42) instead of 42n" or similar)